web/src/billing/upgrade.ts

```diff
@@ -0,0 +1,104 @@
+import $ from "jquery";
+import {z} from "zod";
+
+import * as helpers from "./helpers";
+import {page_params} from "./page_params";
+
+type Prices = {
+    monthly: number;
+    annual: number;
+};
+
+type StripeSessionResponse = {
+    stripe_session_url: string;
+};
+
+type DiscountDetails = {
+    opensource: string;
+    research: string;
+    nonprofit: string;
+    event: string;
+    education: string;
+    education_nonprofit: string;
+};
```

Comment:
Instead of copying over these types on both `upgrade.ts` and `helpers.ts`, it is better to just export these types from `helpers.ts` and just use them here.